File: src/compute_server.cpp
#include <ros/ros.h>
#include "tsu200/Compute.h"

bool handle_compute(tsu200::Compute::Request &req, tsu200::Compute::Response &res) {
  ROS_INFO("Compute request: %d + %d", req.a, req.b);
  res.sum = req.a + req.b;
  return true;
}

int main(int argc, char **argv) {
  ros::init(argc, argv, "compute_server_cpp");
  ros::NodeHandle nh;
  ros::ServiceServer service = nh.advertiseService("compute", handle_compute);
  ROS_INFO("Ready to compute (C++)");
  ros::spin();
  return 0;
}

File: scripts/compute_server.py
#!/usr/bin/env python
import rospy
from tsu200.srv import Compute, ComputeResponse


def handle_compute(req):
    rospy.loginfo("Compute request: %d + %d", req.a, req.b)
    return ComputeResponse(req.a + req.b)


def compute_server():
    rospy.init_node('compute_server')
    s = rospy.Service('compute', Compute, handle_compute)
    rospy.loginfo("Ready to compute")
    rospy.spin()


if __name__ == '__main__':
    compute_server()
